packages/date-picker/src/shared/components/DateInput/DateInputSegment/DateInputSegment.spec.tsx

```diff
@@ -210,34 +210,34 @@ describe('packages/date-picker/shared/date-input-segment', () => {
           onChange: onChangeHandler,
         });
 
-        userEvent.type(input, '6');
+        userEvent.type(input, '4');
         expect(onChangeHandler).toHaveBeenCalledWith(
-          expect.objectContaining({ value: '6' }),
+          expect.objectContaining({ value: '4' }),
         );
       });
     });
   });
 
   describe('Keyboard', () => {
     describe('Backspace', () => {
-      test('clears the input when there is a value', () => {
+      test('does not call the onChangeHandler when the value is initially empty', () => {
         const { input } = renderSegment({
-          value: '26',
           onChange: onChangeHandler,
         });
 
         userEvent.type(input, '{backspace}');
-        expect(onChangeHandler).toHaveBeenCalledWith(
+        expect(onChangeHandler).not.toHaveBeenCalledWith(
```

Comment:
not sure we need `.not.toHaveBeenCalledWith`. Could just have `.not.toHaveBeenCalled`